blocks/form/form.js

```diff
@@ -0,0 +1,433 @@
+function formatNumberPromise(inputs, format) {
+  let arrayInputs = inputs;
+  if (!(inputs instanceof Array)) {
+    arrayInputs = [inputs];
+  }
+  if (!format) {
+    return Promise.resolve(arrayInputs);
+  }
+  function formatFn(formatters) {
+    return arrayInputs.map((num) => {
+      try {
+        return formatters.default(num, format);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log(`unable to format ${num} with format ${format}`);
+        return num;
+      }
+    });
+  }
+  function errorFn(e) {
+    // eslint-disable-next-line no-console
+    console.log('error in obtaining formatters', e);
+    return arrayInputs;
+  }
+  return import('./formatting.js').then(formatFn, errorFn);
```

Comment:
I think the import can be global, as it doesn't seem to be large. but this is a bit a confusing call.
or:

```suggestion
  const formatNumber = import('./formatting.js');
  return formatNumber.then(formatFn, errorFn);
```